src/map/unit.cpp

```diff
@@ -149,29 +165,53 @@ bool unit_walktoxy_nextcell(block_list& bl, bool sendMove, t_tick tick) {
 	if (ud == nullptr)
 		return true;
 
-	int32 speed;
-
 	// Reached end of walkpath
 	if (ud->walkpath.path_pos >= ud->walkpath.path_len)
 		return false;
 
+	// Monsters first check for a chase skill and if they didn't use one if their target is in range each cell after checking for a chase skill
+	if (bl.type == BL_MOB) {
+		mob_data* md = reinterpret_cast<mob_data*>(&bl);
+		// Walk skills are triggered regardless of target due to the idle-walk mob state.
+		// But avoid triggering when already reached the end of the walkpath.
+		// Monsters use walk/chase skills every second, but we only get here every "speed" ms
+		// To make sure we check one skill per second on average, we substract half the speed as ms
+		if (!ud->state.force_walk &&
+			DIFF_TICK(tick, md->last_skillcheck) > MOB_SKILL_INTERVAL - md->status.speed / 2 &&
+			mobskill_use(md, tick, -1)) {
+			if ((ud->skill_id != NPC_SPEEDUP || md->trickcasting == 0) //Stop only when trickcasting expired
+				&& ud->skill_id != NPC_EMOTION && ud->skill_id != NPC_EMOTION_ON //NPC_EMOTION doesn't make the monster stop
+				&& md->state.skillstate != MSS_WALK) //Walk skills are supposed to be used while walking
+			{
+				// Skill used, abort walking
+				// Fix position as walk has been cancelled.
+				clif_fixpos(bl);
+				// Movement was initialized, so we need to return true even though it was stopped
+				// Monsters only start moving or drop target when they stop using chase skills that stop them
+				return true;
+			}
+			// Resend walk packet for proper Self Destruction display
+			sendMove = true;
+		}
+		if (ud->target_to != 0) {
+			int16 tx = ud->to_x;
+			int16 ty = ud->to_y;
+			// Monsters update their chase path one cell before reaching their final destination
+			if (unit_update_chase(bl, tick, (ud->walkpath.path_pos == ud->walkpath.path_len - 1)))
+				return true;
+			// Continue moving, restore to_x and to_y
+			ud->to_x = tx;
+			ud->to_y = ty;
+		}
+	}
+
+	// Get current speed
+	int32 speed;
```

Comment:
I know it is not scope of this PR, but since you moved it down here:
Could be changed to speed = status_get_speed(&bl);
And inside the if just make it speed *=